### PR TITLE
Add unit tests for org.antlr.v4.misc.Utils

### PR DESCRIPTION
**Pull request by EricHetti**
I've analysed your codebase and noticed that `org.antlr.v4.misc.Utils` is not fully tested.
I've written some tests for the methods in this class with the help of [Diffblue Cover](https://www.diffblue.com/opensource).

Hopefully, these tests will help you detect any regressions caused by future code changes. If you would find it useful to have additional tests written for this repository, I would be more than happy to look at other classes that you consider important.

### CODE REVIEW

File: tool-testsuite/test/org/antlr/v4/test/tool/UtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+		Assert.assertNull(Utils.find(new ArrayList<>(), String.class));
+	}
+
+	@Test
+	public void testIndexOf() {
+		ArrayList<String> strings = new ArrayList<>();
+		strings.add("foo");
+		strings.add("bar");
+		Utils.Filter filer = new Utils.Filter() {
```
Comment (petersear) [inline tool-testsuite/test/org/antlr/v4/test/tool/UtilsTest.java]: I'm guessing the variable should be `filter`?

File: tool-testsuite/test/org/antlr/v4/test/tool/UtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+			@Override
+			public boolean select(Object o) {
+				return true;
+			}
+		};
+		Assert.assertEquals(0, Utils.indexOf(strings, filer));
```
Comment (petersear) [inline tool-testsuite/test/org/antlr/v4/test/tool/UtilsTest.java]: `filer` should be `filter`?

File: tool-testsuite/test/org/antlr/v4/test/tool/UtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+		Assert.assertEquals(-1, Utils.indexOf(new ArrayList<>(), null));
+	}
+
+	@Test
+	public void testLastIndexOf() {
+		ArrayList<String> strings = new ArrayList<>();
+		strings.add("foo");
+		strings.add("bar");
+		Utils.Filter filer = new Utils.Filter() {
```
Comment (petersear) [inline tool-testsuite/test/org/antlr/v4/test/tool/UtilsTest.java]: `filer` should be `filter`?

File: tool-testsuite/test/org/antlr/v4/test/tool/UtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+			@Override
+			public boolean select(Object o) {
+				return true;
+			}
+		};
+		Assert.assertEquals(1, Utils.lastIndexOf(strings, filer));
```
Comment (petersear) [inline tool-testsuite/test/org/antlr/v4/test/tool/UtilsTest.java]: `filer` should be `filter`?

File: tool-testsuite/test/org/antlr/v4/test/tool/UtilsTest.java [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+		Assert.assertEquals(-1, Utils.lastIndexOf(new ArrayList<>(), null));
+	}
+
+	@Test
+	public void testSetSize() {
+		ArrayList<String> strings = new ArrayList<>();
```
Comment (petersear) [inline tool-testsuite/test/org/antlr/v4/test/tool/UtilsTest.java:118]: Can we assert side effects here? eg. check the values were set.

**Review by petersear — CHANGES_REQUESTED** [review]

File: tool-testsuite/test/org/antlr/v4/test/tool/UtilsTest.java (hunk 1 above)
Comment (Braavos6) [reply tool-testsuite/test/org/antlr/v4/test/tool/UtilsTest.java]: oops..

Comment (Braavos6) [reply tool-testsuite/test/org/antlr/v4/test/tool/UtilsTest.java]: done

File: tool-testsuite/test/org/antlr/v4/test/tool/UtilsTest.java (hunk 2 above)
Comment (Braavos6) [reply tool-testsuite/test/org/antlr/v4/test/tool/UtilsTest.java]: done

File: tool-testsuite/test/org/antlr/v4/test/tool/UtilsTest.java (hunk 3 above)
Comment (Braavos6) [reply tool-testsuite/test/org/antlr/v4/test/tool/UtilsTest.java]: done

File: tool-testsuite/test/org/antlr/v4/test/tool/UtilsTest.java (hunk 4 above)
Comment (Braavos6) [reply tool-testsuite/test/org/antlr/v4/test/tool/UtilsTest.java]: done

File: tool-testsuite/test/org/antlr/v4/test/tool/UtilsTest.java (hunk 5 above)
Comment (Braavos6) [reply tool-testsuite/test/org/antlr/v4/test/tool/UtilsTest.java:118]: Done

**Review by petersear — APPROVED** [review]

**Review by paulkennethkent — APPROVED** [review]

**Pull request closed without merging**